### PR TITLE
Modify tests to run on zero ports systems 

### PR DESCRIPTION
**Pull request by slutati1536**
Signed-off-by: Sharon Lutati <slutati@nvidia.com>

### Description of PR

Allow systems which starts with zero ports to run platform tests without failing even if the tests skip the interface checks.
- Change the port verification to relay on config_db.json instead of minigraph.xml
- Change places where test fail in case of zero ports on setup

Summary:
Modify tests so that they will pass also on setup with zero ports

### Type of change
- Add checks if PORT exists
- Take port list from config_db.json instead of minigraph

#### What is the motivation for this PR?
Allow systems which starts with zero ports to run platform tests without failing even if the tests skip the interface checks.

#### How did you do it?
Rerun pre defined list of tests where system is defined with zero ports and modify tests one by one to pass but to keep functional as possible (taking into account that interfaces are not configured)

#### How did you verify/test it?
Rerun tests and ensure they are passing on a system with ports and with zero ports.

#### Any platform specific information?
No

#### Supported testbed topology if it's a new test case?
T0 and T1-lag pre defined list of tests

### CODE REVIEW

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@slutati1536 please handle new LGTM errors

**Comment by liat-grozovik** [comment]
@wangxin would you mind to help review the changes?

File: tests/snmp/test_snmp_phy_entity.py [hunk 1]
```diff
@@ -614,7 +614,7 @@ def test_turn_off_psu_and_check_psu_info(duthosts, enum_rand_one_per_hwsku_hostn
     pdu_controller.turn_off_outlet(first_outlet)
     assert wait_until(30, 5, check_outlet_status, pdu_controller, first_outlet, False)
     # wait for psud update the database
-    assert wait_until(120, 20, _check_psu_status_after_power_off, duthost, localhost, creds_all_duts)
+    assert wait_until(180, 20, _check_psu_status_after_power_off, duthost, localhost, creds_all_duts)
```
Comment (liat-grozovik) [inline tests/snmp/test_snmp_phy_entity.py:617]: @slutati1536 what is the motivation to increase timeout?

File: tests/telemetry/test_telemetry.py [hunk 2]
```diff
@@ -153,21 +154,25 @@ def test_telemetry_enabledbydefault(duthosts, rand_one_dut_hostname):
             status_expected = "enabled";
             pytest_assert(str(v) == status_expected, "Telemetry feature is not enabled")
 
+
 def test_telemetry_ouput(duthosts, rand_one_dut_hostname, ptfhost, setup_streaming_telemetry, localhost):
     """Run pyclient from ptfdocker and show gnmi server outputself.
     """
     duthost = duthosts[rand_one_dut_hostname]
+    for asic_index in duthost.get_frontend_asic_ids():
```
Comment (liat-grozovik) [inline tests/telemetry/test_telemetry.py:162]: @slutati1536 please add comment: there is a point to test telemetry only if there is data port. in a case where there is no data port the test should not fail but should also not run.

File: tests/telemetry/test_telemetry.py [hunk 3]
```diff
@@ -234,9 +241,12 @@ def test_virtualdb_table_streaming(duthosts, rand_one_dut_hostname, ptfhost, loc
     duthost = duthosts[rand_one_dut_hostname]
     skip_201911_and_older(duthost)
     cmd = generate_client_cli(duthost=duthost, method=METHOD_SUBSCRIBE, update_count = 3)
-    show_gnmi_out = ptfhost.shell(cmd)['stdout']
-    result = str(show_gnmi_out)
-
-    assert_equal(len(re.findall('Max update count reached 3', result)), 1, "Streaming update count in:\n{0}".format(result))
-    assert_equal(len(re.findall('name: "Ethernet0"\n', result)), 4, "Streaming updates for Ethernet0 in:\n{0}".format(result)) # 1 for request, 3 for response
-    assert_equal(len(re.findall('timestamp: \d+', result)), 3, "Timestamp markers for each update message in:\n{0}".format(result))
+    for asic_index in duthost.get_frontend_asic_ids():
```
Comment (liat-grozovik) [inline tests/telemetry/test_telemetry.py:244]: @slutati1536 same as above. please add comment to the code.

**Review by liat-grozovik — CHANGES_REQUESTED** [review]
@slutati1536 please add for each test logic change the reason why it does not run when there are no interfaces and how the test behaves in this case: cont without failure, pass, ignore, etc.

File: tests/snmp/test_snmp_phy_entity.py (hunk 1 above)
Comment (slutati1536) [reply tests/snmp/test_snmp_phy_entity.py:617]: hey @liat-grozovik, the dut took longer than 120 seconds to update the PSU status, but it passed consistently when the timeout was extended to 180 seconds

File: tests/common/platform/interface_utils.py [hunk 4]
```diff
@@ -109,24 +103,33 @@ def check_interface_information(dut, asic_index, interfaces, xcvr_skip_list):
...
+    port_mapping = cfg_facts.get("PORT", {})
+    for port, port_dict_info in port_mapping.items():
+        port_index = port_dict_info["index"]
+        port_mapping_res[port] = [int(port_index)]
+    return port_mapping_res
+
+
+def validate_config_db_file_exist(dut):
+    config_db_location_cmd = "ls /etc/sonic/config_db.json"
+    config_db_location_string = dut.command(config_db_location_cmd)["stdout"]
```
Comment (wangxin) [inline tests/common/platform/interface_utils.py:133]: Can you use the ansible `state` module to check the existence of a file? It is much easier and more robust than running shell command and searching for specific string in output. Ref: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/stat_module.html

File: tests/platform_tests/mellanox/test_check_sfp_presence.py [hunk 5]
```diff
@@ -13,11 +12,12 @@
     pytest.mark.topology('any')
 ]
 
+
 def test_check_sfp_presence(duthosts, rand_one_dut_hostname, conn_graph_facts):
     """This test case is to check SFP presence status with CLI and sysfs.
     """
     duthost = duthosts[rand_one_dut_hostname]
-    ports_config = json.loads(duthost.command("sudo sonic-cfggen -d --var-json PORT")["stdout"])
+    duthost.command("sudo sonic-cfggen -d --var-json PORT")
```
Comment (wangxin) [inline tests/platform_tests/mellanox/test_check_sfp_presence.py:20]: What's the purpose of running this command here?

**Review by wangxin — COMMENTED** [review]
LGTM except some minor comments. Can you resolve the merge conflict? @slutati1536

File: tests/common/platform/interface_utils.py (hunk 4 above)
Comment (slutati1536) [reply tests/common/platform/interface_utils.py]: @wangxin fixed

**Comment by sujinmkang** [comment]
@slutati1536 Can you please resolve the conflict first?

**Comment by sujinmkang** [comment]
@slutati1536 is there any reason that you decided to go based on config_db.json instead of having an option to choose between config_db.json or minigraph.xml?

**Comment by liat-grozovik** [comment]
@sujinmkang can you please help to review?

**Review by sujinmkang — APPROVED** [review]

File: tests/telemetry/test_telemetry.py (hunk 2 above)
Comment (slutati1536) [reply tests/telemetry/test_telemetry.py]: done

File: tests/telemetry/test_telemetry.py (hunk 3 above)
Comment (slutati1536) [reply tests/telemetry/test_telemetry.py]: done

**Review by wangxin — APPROVED** [review]

**Comment by wangxin** [comment]
@liat-grozovik Currently merging of this PR is blocked by your change request. If you have no concern with this PR, can you approve it? Then we can proceed with merging.

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**